packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -387,6 +391,36 @@ export function getSlideToSegoff(
 	};
 }
 
+const forwardPred = (ref: LocalReferencePosition): boolean =>
+	ref.slidingPreference !== SlidingPreference.BACKWARD;
+const backwardPred = (ref: LocalReferencePosition): boolean =>
+	ref.slidingPreference === SlidingPreference.BACKWARD;
+
+const continueFrom = (node: MergeBlock): boolean => {
+	let siblingExists = false;
+	forwardExcursion(node, () => {
+		siblingExists = true;
+		return false;
+	});
+	return siblingExists;
+};
+
+const onLeaf = (
```

Comment:
nit: I would not remove `onLeaf` and `continueFrom` from their scope. this makes the code harder to read than before, even if it is technically not necessary to create a new function on each execution.

Reply:
if they had informative names perhaps it'd be ok as well.